controllers/authController.ts

```diff
@@ -1 +1,69 @@
-// tutaj dodajemy rejestracje
+import { Request, Response } from 'express';
+import jwt from 'jsonwebtoken';
+import bcrypt from 'bcrypt';
+import { hashPwd } from '../utils/hashPwd';
+import { AdminDb } from '../models/AdminSchema';
+import { UserDb } from '../models/UserSchema';
+import { HrDb } from '../models/HrSchema';
+import { ValidationError } from '../utils/handleError';
+
+export const register = async (req: Request, res: Response) => {
+  const { email, password } = req.body;
+
+  const newUser = new UserDb({
+    email,
+    password: await hashPwd(password),
+    role: 'Kursant',
+  });
+
+  const savedUser = await newUser.save();
+
+  res.status(201).send(savedUser); //@TODO zrobić filtrowanie jak zostanie dodany typ UserRepsonse (bez hasła i tokena)
+};
+
+export const login = async (req: Request, res: Response) => {
+  const { email, password } = req.body;
+
+  let user = await UserDb.findOne({ email });
+  if (!user) user = await HrDb.findOne({ email });
+  if (!user) user = await AdminDb.findOne({ email });
+
+  if (!user) throw new ValidationError('User not found');
+
+  const isMatch = await bcrypt.compare(password, await user.password);
+  if (!isMatch) throw new ValidationError('Wrong password');
+
+  const token = jwt.sign({ id: user._id }, process.env.JWT_SECRET, {
+    expiresIn: 60 * 60 * 24,
```

Comment:
"expiresIn": 
tak się zastanawiam czy nie lepiej to zapisać w jakimś configu czy w zmiennej środowiskowej? Ogólnie zawsze staramy unikać używania takich czystych wartości w funkcji.
W pliku .env mogłoby to wyglądać np. tak JWT_TOKEN_EXPIRATION_TIME= 60 * 60 * 24. Dobrze też stworzyć plik env.example który będziemy pushować do repo (jeżeli jest to jeszcze nie namierzyłem :P), tak żeby każdy na jego podstawie mógł sobie stworzyć lokalnie własny plik .env. Bo widzę, że mamy odnośniki do .env np JWT_SECRET czy SECRET_KEY. 
Np:
JWT_SECRET = jwt_secret
SECRET_KEY = secret_key
JWT_TOKEN_EXPIRATION_TIME = 60 * 60 * 24
itd.